rpc/trace.go

```diff
@@ -0,0 +1,111 @@
+package rpc
+
+import (
+	"fmt"
+
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/jsonrpc"
+	"github.com/NethermindEth/juno/utils"
+	"github.com/NethermindEth/juno/vm"
+)
+
+type TransactionTrace struct {
+	ValidateInvocation    *FunctionInvocation `json:"validate_invocation,omitempty"`
```

Comment:
Do you think we can just use `vm` Types for this?
This looks way too much code to duplicate

Reply:
Unfortunately no, because `CallInfo` has a structure that depends on format of rust-vm (and naming of json fields) and `FunctionInvocation` type is format of json-rpc. For instance:
```
result.Result = info.Execution.Retdata
```
json rpc field `result` is actually `retdata` inside other struct in vm response

Reply:
Can we fix stuff like this in rust types?

For example instead of 

``` rust
#[derive(Debug,Serialize)]
pub struct CallExecution {
    pub retdata: Retdata,
    pub events: Vec<OrderedEvent>,
    pub l2_to_l1_messages: Vec<OrderedL2ToL1Message>,
}
```

can we rename `retdata` to `result` here?

We have 3 almost-exactly same type definitions, 1 in Rust, 2 in Go. If we can at least get rid of one of them. It would be already an huge improvement.

Reply:
I wouldnt mind even if we straight up used the JSON that we return from rust. Without unmarshaling/marshaling on Go side.